tests/test_events.py

```diff
@@ -14,3 +15,13 @@ def test_emit(self, requests_mock: Adapter) -> None:
                 request.json() == {'data': {'foo': 'bar'}},
         )
         Events(new_session(api_url='http://api')).emit({'foo': 'bar'})
+
+    def test_emit_with_key(self, requests_mock: Adapter) -> None:
+        requests_mock.register_uri(
+            'POST', 'http+api://api/events',
+            text='OK',
+            request_headers={'content-type': 'application/json'},
+            additional_matcher=lambda request:
+                request.json() == {'data': {'foo': 'bar'}, 'key': 'key'},
+        )
+        Events(new_session(api_url='http://api')).emit({'foo': 'bar'}, "key")
```

Comment:
Typically for keyword arguments you do want to explicitly specify their names, so rather prefer `emit({'foo': 'bar'}, key='key')`.